integrations/graal/native-image-extension/src/main/java/io/helidon/integrations/graal/nativeimage/extension/HelidonReflectionFeature.java

```diff
@@ -742,10 +753,22 @@ boolean isExcluded(Class<?> theClass) {
 
         private final Class<?> clazz;
 
+        private boolean validated;
+        private boolean valid = true;
```

Comment:
How are the two variables different?

Reply:
`validated` marks whether we touched this file
`valid` means it is valid (if `validated` is true as well)
This is to make sure the validation happens before we start registering the class

Reply:
`validated` marks whether we touched this file
`valid` means it is valid (if `validated` is true as well)
This is to make sure the validation happens before we start registering the class